src/release/breaking-changes/boxdecoration-shape-migration.md

```diff
@@ -0,0 +1,108 @@
+---
+title: Migration guide for BoxDecoration.circle
+description: Removal of BoxDecoration.circle and how to migrate
+---
+
+## Summary
+
+This change removes `BoxDecoration(shape: BoxShape.circle)`. Existing apps
+should use `ShapeDecoration(shape: CircleBorder())` instead.
+
+## Context
+
+`BoxDecoration(shape: BoxShape.circle)` uses an enum. Each new enum value requires new code
+in many places. That new code may not shake out,
+since the compiler doesn't know which values never get used.
+`BoxDecoration` has a permanent binary size cost,
+even for people who don't use it.
+Flutter moved to `ShapeBorders` rather than `BoxShape` for new
+shapes, and the existence of `BoxShape` remains as a legacy feature.
+
+Use `ShapeDecoration`. It aligns with the rest of the Flutter API.
+`BoxShape` has been deprecated.
+Continuing to use that feature can cause confusion.
+
+## Description of change
+
+Remove `BoxDecoration(shape: BoxShape.circle)` and `BoxDecoration.shape`.
```

Comment:
What should the user replace it with? Are there other cases of using `BoxShape` that users may need to migrate?

Reply:
No because we are not changing InkWell.